src/common/pony/detail/atomics.h

```diff
@@ -119,17 +117,18 @@ namespace ponyint_atomics
 namespace ponyint_atomics
 {
   template <typename T>
-  inline PONY_ABA_PROTECTED(T) big_load(PONY_ABA_PROTECTED(T)* ptr)
+  inline PONY_ABA_PROTECTED_PTR(T) big_load(PONY_ABA_PROTECTED_PTR(T)* ptr)
   {
-    PONY_ABA_PROTECTED(T) ret = {NULL, 0};
+    PONY_ABA_PROTECTED_PTR(T) ret = {NULL, 0};
```

Comment:
This is because x86 doesn't have plain plain load/store operations on 16 bytes. The only instruction that can work on an operand of that size is `cmpxchg16b` and if we don't use that and use 2 plain `mov` instead, there would be a time window where a thread could modify one part of the value while another thread is still processing the other part.  
For reference, both [GCC and Clang](https://godbolt.org/g/u3mD8t) also generate `cmpxchg16b` for atomic loads/stores of 16 bytes objects.

I've tried to reduce the overhead introduced by these operations by using non-atomic loads/stores whenever possible. For example, [pool.c:465](https://github.com/ponylang/ponyc/pull/1621/files#diff-174c1821e490b93d3d279cc44a9a4268R465). Also, the fast bailout path of both `pool_pull` and `pool_block_pull` doesn't use any big atomic or hardware synchronisation.